tests/test_explainability/test_shap_explainers.py

```diff
@@ -0,0 +1,410 @@
+            if isinstance(explainer, SHAPTreeExplainer) and isinstance(
+                estimator, GradientBoostingClassifier
+            ):
+                # there is currently a bug in SHAP's TreeExplainer for GradientBoostingClassifier
+                # https://github.com/shap/shap/issues/3177 returning only one feature weight
+                # which is also based on log odds. This check is a workaround until the bug is fixed.
+                self.assertEqual(
+                    (nof_features,), explanation.feature_weights.shape  # type: ignore[union-attr]
+                )
+            elif isinstance(estimator, SVC):
+                # SVC seems to be handled differently by SHAP. It returns only a one dimensional
+                # feature array for binary classification.
+                self.assertTrue(
+                    (1,), explanation.prediction.shape  # type: ignore[union-attr]
+                )
+                self.assertEqual(
+                    (nof_features,), explanation.feature_weights.shape  # type: ignore[union-attr]
+                )
+            else:
+                # normal binary classification case
+                self.assertEqual(
+                    (nof_features, 2), explanation.feature_weights.shape  # type: ignore[union-attr]
+                )
+        else:
+            raise ValueError("Error in unittest. Unsupported estimator.")
+
+        if issubclass(type(explainer), AtomExplanationMixin):
+            self.assertIsInstance(explanation.atom_weights, np.ndarray)  # type: ignore[union-attr]
+            self.assertEqual(
+                explanation.atom_weights.shape,  # type: ignore[union-attr]
+                (explanation.molecule.GetNumAtoms(),),  # type: ignore[union-attr]
+            )
+
+    def test_explanations_fingerprint_pipeline(  # pylint: disable=too-many-locals
+        self,
+    ) -> None:
+        """Test SHAP's TreeExplainer wrapper on MolPipeline's pipelines with fingerprints."""
+
+        tree_estimators = [
+            RandomForestClassifier(n_estimators=2, random_state=_RANDOM_STATE),
+            RandomForestRegressor(n_estimators=2, random_state=_RANDOM_STATE),
+            GradientBoostingClassifier(n_estimators=2, random_state=_RANDOM_STATE),
+            GradientBoostingRegressor(n_estimators=2, random_state=_RANDOM_STATE),
+        ]
+        other_estimators = [
+            SVC(kernel="rbf", probability=False, random_state=_RANDOM_STATE),
+            SVR(kernel="linear"),
+            LogisticRegression(random_state=_RANDOM_STATE),
+            LinearRegression(),
+        ]
+        n_bits = 64
+
+        explainer_types = [
+            SHAPKernelExplainer,
+            SHAPTreeExplainer,
+        ]
+        explainer_estimators = [tree_estimators + other_estimators, tree_estimators]
+
+        for estimators, explainer_type in zip(explainer_estimators, explainer_types):
```

Comment:
maybe make sure that len(explainer_estimators) == len(explainer_types). Otherwise extending the test might break things.

Reply:
Done. I added `strict=True`